### PR TITLE
Updated flash constants

### PR DESCRIPTION
**Pull request by ShinyTillDawn**
Please let me know if I did this in the wrong order. I've never done PRs before.
I updated the table related to the move Flash and added a constant to go along with it.

### CODE REVIEW

File: src/HexManiac.Core/Models/Code/constantReference.txt [hunk 1]
```diff
@@ -388,3 +388,10 @@ BPRE0.data.abilities.pickup.length-2 02CEEC
 BPGE0.data.abilities.pickup.length-2 02CEEC
 BPRE1.data.abilities.pickup.length-2 02CF00
 BPGE1.data.abilities.pickup.length-2 02CF00
+
+# maximum flash level
+BPEE0.scripts.moves.flash.maxlevel-1 0854E8
+BPRE0.scripts.moves.flash.maxlevel-1 055D04
+BPGE0.scripts.moves.flash.maxlevel-1 055D04
+AXVE0.scripts.moves.flash.maxlevel-1 053CE0
```
Comment (haven1433) [inline src/HexManiac.Core/Models/Code/constantReference.txt:396]: Please update to 39ACE8, 39AB30, 39AD00, and 39AB48 for Ruby 1.0, Sapphire 1.0, Ruby 1.1, and Sapphire 1.1

File: src/HexManiac.Core/Models/Code/constantReference.txt [hunk 2]
```diff
@@ -388,3 +388,10 @@ BPRE0.data.abilities.pickup.length-2 02CEEC
 BPGE0.data.abilities.pickup.length-2 02CEEC
 BPRE1.data.abilities.pickup.length-2 02CF00
 BPGE1.data.abilities.pickup.length-2 02CF00
+
+# maximum flash level
+BPEE0.scripts.moves.flash.maxlevel-1 0854E8
+BPRE0.scripts.moves.flash.maxlevel-1 055D04
```
Comment (haven1433) [inline src/HexManiac.Core/Models/Code/constantReference.txt:394]: 3C68E0,  3C671C, 3C6950, 3C678C for FireRed 1.0, Leafgreen 1.0, FireRed 1.1, and LeafGreen 1.1

File: src/HexManiac.Core/Models/Code/constantReference.txt [hunk 3]
```diff
@@ -388,3 +388,10 @@ BPRE0.data.abilities.pickup.length-2 02CEEC
 BPGE0.data.abilities.pickup.length-2 02CEEC
 BPRE1.data.abilities.pickup.length-2 02CF00
 BPGE1.data.abilities.pickup.length-2 02CF00
+
+# maximum flash level
+BPEE0.scripts.moves.flash.maxlevel-1 0854E8
```
Comment (haven1433) [inline src/HexManiac.Core/Models/Code/constantReference.txt:393]: Should be 54FE78

File: src/HexManiac.Core/Models/Code/tableReference.txt [hunk 4]
```diff
@@ -198,7 +198,7 @@ data.pokemon.moves.tutorcompatibility,       ,       ,       ,       , 120C30, 1
 data.pokemon.moves.tms,                06F038, 06F03C, 06F058, 06F05C, 125A8C, 125A64, 125B04, 125ADC, 1B6D10, [move:data.pokemon.moves.names]58
 data.pokemon.moves.tmcompatibility,    0403B0, 0403B0, 0403D0, 0403D0, 043C68, 043C68, 043C7C, 043C7C, 06E048, [moves|b[]data.pokemon.moves.tms]data.pokemon.names
 data.pokemon.moves.hms,                040A24, 040A24, 040A44, 040A44, 0441DC, 0441DC, 0441F0, 0441F0, 06E828, [move:data.pokemon.moves.names]8
-data.pokemon.moves.details.flash.radius,       0815DC, 0815DC, 0815FC, 0815FC, 07F070, 07F080, 07F084, 07F058, 0B00E4, [max: current:]1
+data.pokemon.moves.details.flash.radius,       0815DC, 0815DC, 0815FC, 0815FC, 07F070, 07F080, 07F084, 07F058, 0B00E4, [current:]scripts.moves.flash.maxlevel
```
Comment (haven1433) [inline src/HexManiac.Core/Models/Code/tableReference.txt:201]: So I guess the idea is that there's 5 values? I previously had it listed as a pair of values, 2 bytes each. You've changed it to a single value, 2 bytes. Looking at the data, I think there's 5 pairs of 2 byte values (20 bytes total), not just 5 values where each is 2 bytes (10 bytes total). I'm guessing it's like an x/y pair or something. Are you sure it's just `current:` and not something like `x: y:`?

**Review by haven1433 — COMMENTED** [review]
I think you've misunderstood how `contstantReference.txt` works. While `tableReference.txt` wants the address of pointers, `constantReference.txt` wants the actual address of the constants. This is because most constants are part of assembly code, and don't actually have anything pointing to them.

Also, in order for a feature to be added to HMA, it has to include all 9 supported games *(there's a few exceptions, such as tutors, since the tutor feature is actually missing from R/S)*. I've found the addresses for you for R/S/FR/LG 1.1, please add them to your PR.

File: src/HexManiac.Core/Models/Code/tableReference.txt (hunk 4 above)
Comment (ShinyTillDawn) [reply src/HexManiac.Core/Models/Code/tableReference.txt:201]: This is what it says in pret pokéemerald:
`// Smaller flash level -> larger flash radius`
`static const u16 sFlashLevelToRadius[] = { 200, 72, 64, 56, 48, 40, 32, 24, 0 };`
`const s32 gMaxFlashLevel = ARRAY_COUNT(sFlashLevelToRadius) - 1;`

Based on this, I'm pretty sure there are 9 values that correspond to levels 0−8.

**Comment by ShinyTillDawn** [comment]
I just made a new commit that hopefully resolves my blunders.

**Pull request merged**